File: react-java/backend/src/main/java/com/clt/hrm/core/administration/program/dtos/SearchProgramDto.java
package com.clt.hrm.core.administration.program.dtos;

import com.clt.hrm.infra.common.dtos.PaginationDto;
import com.clt.hrm.infra.common.dtos.SortDto;

import lombok.Data;

@Data
public class SearchProgramDto {
	private String coId;
	private String pgmId;
	private String pgmCd;
    private String pgmNm;
    private String pgmTpCd;
    private String useFlg;
    
    // Pagination & Sorting
    private SortDto sort;
    private PaginationDto pagination;
}

File: nestjs/backend/src/apps/administration/programs/dto/search-program.dto.ts
import { IsOptional, IsString } from 'class-validator';
import { SearchBaseDto } from '@infra/common/dtos';

export class SearchProgramDto extends SearchBaseDto {
  @IsOptional()
  @IsString()
  pgmId?: string;

  @IsOptional()
  @IsString()
  pgmCd?: string;

  @IsOptional()
  @IsString()
  pgmNm?: string;

  @IsOptional()
  @IsString()
  pgmTpCd?: string;
}
